src/ionosphere/waccmx/edyn_esmf.F90

```diff
@@ -389,30 +293,42 @@ subroutine edyn_esmf_init( mpi_comm )
       integer              :: petCount
       integer              :: iam
       integer              :: npes
-      integer, allocatable :: petlist(:)
+      integer              :: mag_npes !!XXgoldyXX: get this from mag grid?
+      type(ESMF_DELayout)  :: geo_delayout
+      type(ESMF_DELayout)  :: mag_delayout
       type(ESMF_VM)        :: vm_init
-      type(ESMF_VM)        :: vm_curr
-
-      type(ESMF_GridComp)  :: phys_comp ! physics grid mesh
+#endif
+!!XXgoldyXX: ^ remove?
+      character(len=*), parameter :: subname = 'edyn_esmf_init'
 
+!!XXgoldyXX: v remove?
+#if 0
+      !! Create distribution and PE information for each grid
       call ESMF_VMGetCurrent(vm_init, rc=rc)
-      call edyn_esmf_chkerr('ESMF_VMGetCurrent', rc)
+      call edyn_esmf_chkerr(subname, 'ESMF_VMGetCurrent', rc)
       call ESMF_VMGet(vm_init, localPet=localPet, petCount=petCount)
-      call edyn_esmf_chkerr('ESMF_VMGet', rc)
+      call edyn_esmf_chkerr(subname, 'ESMF_VMGet', rc)
       call mpi_comm_size(mpi_comm, npes, rc)
       call mpi_comm_rank(mpi_comm, iam, rc)
-
-      ! Collect all the PETS for each instance
+      ! Collect all the PETS for each instance for phys grid
       allocate(petlist(npes))
       call mpi_allgather(localPet, 1, MPI_INTEGER, petlist, 1, &
            MPI_INTEGER, mpi_comm, rc)
-      ! Now, we should be able to create a gridded component with these pets
-      phys_comp = ESMF_GridCompCreate(petList=petList, name=trim(inst_name), rc=rc)
-      call edyn_esmf_chkerr('ESMF_GridCompCreate '//trim(inst_name), rc)
-      call ESMF_GridCompSetServices(phys_comp, edyn_gcomp_SetServices, rc=rc)
-      call edyn_esmf_chkerr('ESMF_GridCompSetServices '//trim(inst_name), rc)
-      call ESMF_GridCompInitialize(phys_comp, rc=rc)
-      call edyn_esmf_chkerr('ESMF_GridCompRun '//trim(inst_name), rc)
+      phys_delayout = ESMF_DELayoutCreate(deCount=npes, petList=petlist, rc=rc)
+      call edyn_esmf_chkerr(subname, 'phys ESMF_DELayoutCreate', rc)
+      geo_delayout = ESMF_DELayoutCreate(deCount=geo_npes,                    &
+           petList=petlist(1:geo_npes), rc=rc)
+      call edyn_esmf_chkerr(subname, 'geo ESMF_DELayoutCreate', rc)
+      geo_distgrid = ESMF_DistGridCreate(minIndex=(/1/),                      &
+           maxIndex=(/geo_npes/), delayout=geo_delayout, rc=rc)
+      call edyn_esmf_chkerr(subname, 'geo ESMF_DistGridCreate', rc)
+      mag_npes = npes !!XXgoldyXX: get this from mag grid?
```

Comment:
I think we let mag_npes be equal to geo_npes.   Otherwise, we might need to do some major refactoring of edyn_mpi module.

Reply:
Okay.